cpp/src/arrow/ipc/test_common.cc

```diff
@@ -275,6 +275,20 @@ Status MakeIntRecordBatch(std::shared_ptr<RecordBatch>* out) {
   return MakeIntBatchSized(10, out);
 }
 
+Status MakeIntBatchWithMetadata(std::shared_ptr<KeyValueMetadata> key_value_metadata,
+                                std::shared_ptr<RecordBatch>* out) {
```

Comment:
I add a new test utility function to generate a random int batch with the provided metadata key value pairs.